torchao/float8/fsdp_utils.py

```diff
@@ -84,6 +84,8 @@ def precompute_float8_dynamic_scale_for_fsdp(module: nn.Module) -> None:
     torch.ops.aten.as_strided.default,
     torch.ops.aten._to_copy.default,
     torch.ops.aten._pin_memory.default,
+    torch.ops.aten.split.Tensor,
```

Comment:
`torch.chunk` are called in both TP (`distribute_tensor`) and FSDP2

For FSDP2, preserving `WeightWithDynamicFloat8CastTensor` still work because we only call `.size()` after `torch.chunk`  

https://github.com/pytorch/pytorch/blob/2b86a7fcc78eac8510e0f912b9287107af47d50a/torch/distributed/_composable/fsdp/_fsdp_param.py#L310-L312

During TP init, it's critical for `torch.chunk` to preserve `WeightWithDynamicFloat8CastTensor`. FSDP2 init happens after TP init and FSDP2 need to see `DTensor(local=WeightWithDynamicFloat8CastTensor)`

https://github.com/pytorch/pytorch/blob/2b86a7fcc78eac8510e0f912b9287107af47d50a/torch/distributed/_composable/fsdp/_fsdp_param.py#L289